### ISSUE TITLE
When I accelerated the model, SOMETHING went wrong

### ISSUE THREAD

**Issue by hxzl-98** [microsoft/nni#4870]
**Describe the issue**:
An error is reported when the following code is run：
ModelSpeedup(model, torch.rand(3, 1, 28, 28).to(device), masks).speedup_model()

Error message is：
[ CPUFloatType{8732,4} ]) of traced region did not have observable data dependence with trace inputs; this probably indicates your program cannot be understood by the tracer.

The question that has puzzled me for a long time, hope to get the answer

**Environment**:
- NNI version:
- Training service (local|remote|pai|aml|etc):
- Client OS:
- Server OS (for remote mode only):
- Python version:
- PyTorch/TensorFlow version:
- Is conda/virtualenv/venv used?:
- Is running in Docker?:

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:

**Comment by J-shang** [comment]
hello @hxzl-98 , I think maybe this is because your model can not work well with `torch.jit.trace`, could you show your model for us to reproduce your issue?

**Comment by hxzl-98** [comment]
hello @J-shang ,I used the open source project SSD-PyTorch's model, structured as follows:

class SSD(nn.Module):
    
    def __init__(self, phase, size, base, extras, head, num_classes):
        super(SSD, self).__init__()
        self.phase = phase
        self.num_classes = num_classes
        self.cfg = (coco, voc)[num_classes == 3]
        self.priorbox = PriorBox(self.cfg)
        self.priors = Variable(self.priorbox.forward(), volatile=True)
        self.size = size

        # SSD network
        self.vgg = nn.ModuleList(base)
        # Layer learns to scale the l2 normalized features from conv4_3
        self.L2Norm = L2Norm(512, 20)
        self.extras = nn.ModuleList(extras)

        self.loc = nn.ModuleList(head[0])
        self.conf = nn.ModuleList(head[1])

        if phase == 'test':
            self.softmax = nn.Softmax(dim=-1)
            self.detect = Detect(num_classes, 0, 200, 0.01, 0.45)

    def forward(self, x):
        sources = list()
        loc = list()
        conf = list()

        # apply vgg up to conv4_3 relu
        for k in range(23):
            x = self.vgg[k](x)

        s = self.L2Norm(x)
        sources.append(s)

        # apply vgg up to fc7
        for k in range(23, len(self.vgg)):
            x = self.vgg[k](x)
        sources.append(x)

        # apply extra layers and cache source layer outputs
        for k, v in enumerate(self.extras):
            x = F.relu(v(x), inplace=True)
            if k % 2 == 1:
                sources.append(x)

        # apply multibox head to source layers
        for (x, l, c) in zip(sources, self.loc, self.conf):
            loc.append(l(x).permute(0, 2, 3, 1).contiguous())
            conf.append(c(x).permute(0, 2, 3, 1).contiguous())

        loc = torch.cat([o.view(o.size(0), -1) for o in loc], 1)
        conf = torch.cat([o.view(o.size(0), -1) for o in conf], 1)
        if self.phase == "test":
            
            output=self.detect.forward(loc.view(loc.size(0), -1, 4),
                                       self.softmax(conf.view(conf.size(0), -1, self.num_classes)),
                                       self.priors.type(type(x.data)))
        else:
            output = (
                loc.view(loc.size(0), -1, 4),
                conf.view(conf.size(0), -1, self.num_classes),
                self.priors
            )
        return output
****************************************
Thanks for your help

**Comment by J-shang** [comment]
seems `self.priors` have no relationship with forward `input`? remove `self.priors` from output maybe can solve your problem.

FYI, a related issue https://github.com/pytorch/pytorch/issues/31591

**Comment by hxzl-98** [comment]
I solved it. Thank you for your help

**Comment by jdcmj** [comment]
> I solved it. Thank you for your help

Excuse me ,I meet the same problem ,how did you solve it?

**Comment by gkrisp98** [comment]
Hi @hxzl-98, could you please share the way you solved this problem because I am facing a similar one
